src/jit/gentree.h

```diff
@@ -4209,6 +4215,42 @@ struct GenTreeSIMD : public GenTreeOp
 };
 #endif // FEATURE_SIMD
 
+#if defined(_TARGET_XARCH_) && !defined(LEGACY_BACKEND)
+struct GenTreeHWIntrinsic : public GenTreeOp
+{
+    NamedIntrinsic gtHWIntrinsicId;
+    var_types      gtBaseType; // SIMD vector base type
+    unsigned       gtSIMDSize; // SIMD vector size in bytes, use 0 for scalar intrinsics
```

Comment:
Maybe it should be `unsigned char`? `var_types` is so you get 3 bytes of padding between `gtBaseType` and `gtSIMDSize`.

Reply:
Good point, I will keep this in mind and try to optimize here when I implement SIMD intrinsics.